### DISCUSSION TITLE
Avatars & Badges

### DISCUSSION THREAD

**Discussion by jacywest** [MCLD/greatreadingadventure#1006] [Help!]
I know this is probably simple stuff to ask about, but our IT is currently knee-deep in other projects so I'm trying to get what I can done on my own before summer and this is my first time working on something like this. 

Avatars - I've been looking at the downloadable files from [Glitch](https://www.glitchthegame.com/public-domain-game-art/) and downloaded a bunch of them to upload into the Avatars tab of the GRA in Mission Control, but they give an error or most of them say the file is too large. How do I get avatars set up?

Badges - I figured out the site to create and manage badges (openbadges.me, unless I'm completely off track) but once badges are made, I have no idea how to connect that piece to the GRA site so that badges appear in there. So I currently can't make any challenges or secret codes. Is that something I can do from Mission Control in the site or are there directions I can send to my IT so they can do it in the back end?

I've read the manual, so I've also been trying to set up templates and icons, but the above is what's keeping me from actually setting up challenges, etc.

**Comment by k7hpn** [comment]
Hello @jacywest - I am so sorry about the extended delay in getting back to you on this. Apparently GitHub has stopped notifying me about discussion posts and I'm only just seeing this now. This is probably of no use to you for 2023 but I'll respond to your questions in case it helps you or other users for future programs.

1. There are currently two ways to get avatars into the program: using the default avatar bundle or crafting a special bundle `.zip` file containing `.json` instructions on how to import the avatar files present in the `.zip` archive. The most straightforward aproach by far is to import the default avatar bundle and then remove any parts that you would prefer to not be available. You are correct - the default avatar bundle is too large to upload using the upload interface. Here are two ways to get the default bundle imported:

   1. If you installed the GRA from the release `.zip` file you should see a "Add default avatars" button when you visit the Avatars area of Mission Control.
   2. If you do not see the button, you can [download the latest avatar `.zip` file](https://github.com/MCLD/gra-avatars/releases) and place it in `shared/private/` named `avatars.zip` - then you should see an "Add avatars.zip" button in the Avatars area of Mission Control.

2. Badges are generally supplied attached to another feature (e.g. when creating a Challenge or as a sign-up award in Program configuration). There are two ways that this can work:
   1. In Site Management under the Settings tab you are able to toggle on the integrated link to the openbadges.me badge maker. When it is enabled, you can launch it directly from any place that you can assign a badge in the software, create your badge in the pop-up, and then save it. The pop-up disappears and it should be available in the software.
   2. Whether or not the integration is enabled, you can always upload a square image as a badge and the software will handle resizing it and attaching it. These can be square images from openbadges.me or any other image source.

I hope this helps and I apologize again for the extreme delay. :(

**Reply by jholder007** [reply]
Same problem with avatars:

Defaults would not load, and adding the avatars.zip to the shared/private folder resulted in the add default avatars button changing to import avatars.zip, with the same result: Unable to connect to the server to run the job.

Help!
![2024-03-09_20-40-21](https://github.com/MCLD/greatreadingadventure/assets/162853392/85685003-6fd7-4bee-a396-80184501f20e)
![2024-03-09_20-40-39](https://github.com/MCLD/greatreadingadventure/assets/162853392/aa16b257-7f95-470b-82f7-c24e6c80e3af)

**Reply by k7hpn** [reply]
This sounds to me like it's the job system that is failing due to an issue with Web Sockets. Are you able to successfully run any reports or do they give you the same error?

Also, if it's writing out a log file, is there any more information about the error there?
